LeBlue/src/main/java/houtbecke/rs/le/fourdotthree/LeDevice43.java

```diff
@@ -215,44 +215,43 @@ public void startScanning() {
     }
 
     @Override
-    public void startScanning(UUID... uuids) {
-        if(!hasPermission() )return;
-        BluetoothLeScannerCompat scanner = BluetoothLeScannerCompat.getScanner();
-        ScanSettings settings = new ScanSettings.Builder()
-                .setScanMode(ScanSettings.SCAN_MODE_BALANCED).setNumOfMatches(ScanSettings.MATCH_NUM_FEW_ADVERTISEMENT)
-                .setUseHardwareBatchingIfSupported(false).build();
-        List<ScanFilter> filters = new ArrayList<>();
-        ScanFilter.Builder builder = new ScanFilter.Builder();
-        for (UUID uuid : uuids ){
-            builder  = builder.setServiceUuid(new ParcelUuid(uuid));
+    public void startScanning(final UUID... uuids) {
+        if(!hasPermission()) {
+            return;
+        }
 
+        final BluetoothLeScannerCompat scanner = BluetoothLeScannerCompat.getScanner();
+        final List<ScanFilter> scanFilters = new ArrayList<>();
+        for(UUID uuid : uuids) {
+            scanFilters.add(new ScanFilter.Builder().setServiceUuid(new ParcelUuid(uuid)).build());
         }
-        filters.add(new ScanFilter.Builder().build());
-        scanner.startScan(filters, settings, scanCallback);
+        startScanning(scanner, scanFilters);
     }
 
     @Override
-    public void startScanning(List<List<UUID>> filters) {
-        if(!hasPermission() )return;
-        BluetoothLeScannerCompat scanner = BluetoothLeScannerCompat.getScanner();
-        ScanSettings settings = new ScanSettings.Builder()
-                .setScanMode(ScanSettings.SCAN_MODE_BALANCED).setNumOfMatches(ScanSettings.MATCH_NUM_FEW_ADVERTISEMENT)
-                .setUseHardwareBatchingIfSupported(false).build();
-        List<ScanFilter> scanFilters = new ArrayList<>();
-        for (List<UUID> filter : filters){
-            ScanFilter.Builder builder = new ScanFilter.Builder();
-            for (UUID uuid : filter ){
-                builder  = builder.setServiceUuid(new ParcelUuid(uuid));
+    public void startScanning(final List<List<UUID>> filters) {
+        if(!hasPermission()) {
+            return;
+        }
 
+        final BluetoothLeScannerCompat scanner = BluetoothLeScannerCompat.getScanner();
+        final List<ScanFilter> scanFilters = new ArrayList<>();
+        for(List<UUID> filter : filters) {
+            for(UUID uuid : filter) {
+                scanFilters.add(new ScanFilter.Builder().setServiceUuid(new ParcelUuid(uuid)).build());
```

Comment:
you are now creating a new ScanFilter for each uuid.  There should be multiple filters which can contain multiple uuids

Reply:
That's correct, I create one ScanFilter instance per each uuid and add it to a list of filters, which is used by a scanner later on. Current API doesn't allow me to set multiple uuids per single filter.